Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -620,13 +624,14 @@ public override bool TrySplashSpillAt(EntityUid uid,
     /// <inheritdoc/>
     public override bool TrySpillAt(EntityCoordinates coordinates, Solution solution, out EntityUid puddleUid, bool sound = true)
     {
-        if (solution.Volume == 0)
+        if (solution.Volume == 0 || !_xformQuery.TryGetComponent(coordinates.EntityId, out var transform))
         {
             puddleUid = EntityUid.Invalid;
             return false;
         }
 
-        var gridUid = coordinates.GetGridUid(EntityManager);
+        var gridUid = _transformSystem.GetGrid((coordinates.EntityId, transform));
```

Comment:
```suggestion
        var gridUid = _transformSystem.GetGrid(coordinates.EntityId);
```
This will actually use an internal `EntityQuery<TransformComponent>` in `EntityManager`, so it will be fast without having to set up the transform query yourself.